src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorHelpers.cs

```diff
@@ -0,0 +1,641 @@
+				// Otherwise, we'll place it on the boundary that was exceeded.
+				if (valueToIncrement < minBound)
+				{
+					valueToIncrement = (shouldWrap && previousValue == minBound) ? maxBound : minBound;
+				}
+
+				if (valueToIncrement > maxBound)
+				{
+					valueToIncrement = (shouldWrap && previousValue == maxBound) ? minBound : maxBound;
+				}
+
+				// Uno Doc: WinUI/C++ modifies the newHsv channel directly using a pointer to the appropriate channel.
+				// This isn't desirable in C# so once the modification is done it has to be copied back to newHsv.
+				switch (channel)
+				{
+					case ColorPickerHsvChannel.Hue:
+						newHsv.H = valueToIncrement;
+						break;
+
+					case ColorPickerHsvChannel.Saturation:
+						newHsv.S = valueToIncrement;
+						break;
+
+					case ColorPickerHsvChannel.Value:
+						newHsv.V = valueToIncrement;
+						break;
+
+					default:
+						throw new InvalidOperationException("Invalid ColorPickerHsvChannel.");
+				}
+
+				// We multiplied saturation and value by 100 previously, so now we want to put them back in the 0-1 range.
+				newHsv.S /= 100;
+				newHsv.V /= 100;
+			}
+			else
+			{
+				// While working with named colors, we're going to need to be working in actual HSV units,
+				// so we'll divide the min bound and max bound by 100 in the case of saturation or value,
+				// since we'll have received units between 0-100 and we need them within 0-1.
+				if (channel == ColorPickerHsvChannel.Saturation ||
+					channel == ColorPickerHsvChannel.Value)
+				{
+					minBound /= 100;
+					maxBound /= 100;
+				}
+
+				newHsv = FindNextNamedColor(originalHsv, channel, direction, shouldWrap, minBound, maxBound);
+			}
+
+			return newHsv;
+		}
+
+		// Uno Doc: This function is removed and instead Math.Sign() is used.
+		// Implementation of the signum function, which returns the sign of a number while discarding its value.
+		/*template <typename T>
+		int sgn(T val)
+		{
+			const int first = (static_cast<T>(0) < val) ? 1 : 0;
+			const int second = (static_cast<T>(0) > val) ? 1 : 0;
+
+			return first - second;
+		}*/
+
+		public static Hsv FindNextNamedColor(
+			Hsv originalHsv,
+			ColorPickerHsvChannel channel,
+			IncrementDirection direction,
+			bool shouldWrap,
+			double minBound,
+			double maxBound)
+		{
+			// There's no easy way to directly get the next named color, so what we'll do
+			// is just iterate in the direction that we want to find it until we find a color
+			// in that direction that has a color name different than our current color name.
+			// Once we find a new color name, then we'll iterate across that color name until
+			// we find its bounds on the other side, and then select the color that is exactly
+			// in the middle of that color's bounds.
+			Hsv newHsv = originalHsv;
+			/*
+			string originalColorName = ColorHelper.ToDisplayName(ColorConversion.ColorFromRgba(ColorConversion.HsvToRgb(originalHsv)));
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)